pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -205,6 +215,14 @@ func (c *ServiceBuilder) CreateTransformerService(name string, metadata metav1.O
 	if hasInferenceLogging {
 		addLoggerContainerPort(container)
 	}
+
+	predictor := ""
+	if isCanary {
```

Comment:
A similar simplification can be done here.

Reply:
Updated as suggested